terraform/modules/ecs/main.tf

```diff
@@ -1,123 +1,58 @@
-resource "aws_ecs_cluster" "droits-ecs-cluster" {
-  name = var.ecs_cluster_name
-}
-
-
-resource "aws_ecs_task_definition" "backoffice-task-definition" {
-  family                   = "backoffice"
+resource "aws_ecs_task_definition" "task-definition" {
+  family                   = var.service_name
   execution_role_arn       = var.iam_role_arn
   requires_compatibilities = ["FARGATE"]
   network_mode             = "awsvpc"
-  cpu                      = var.backoffice_fargate_cpu
-  memory                   = var.backoffice_fargate_memory
+  cpu                      = var.fargate_cpu
+  memory                   = var.fargate_memory
   container_definitions = jsonencode([{
-    name : "backoffice",
-    image : var.backoffice_image_url,
-    cpu : var.backoffice_fargate_cpu,
-    memory : var.backoffice_fargate_memory,
-    environment : [{ "name" : "ENV_FILE", "value" : "${var.backoffice_environment_file}" }],
+    name : var.service_name,
+    image : var.image_url,
+    cpu : var.fargate_cpu,
+    memory : var.fargate_memory,
+    environment : [{ "name" : "ENV_FILE", "value" : "${var.environment_file}" }],
     portMappings : [
       {
-        containerPort : var.backoffice_port
-        hostPort : var.backoffice_port
+        containerPort : var.port
+        hostPort : var.port
       }
     ],
     healthCheck : {
       retries : 6,
       command : [
-        "CMD-SHELL", "curl -f http://localhost:5000/healthz || exit 1"
+        "CMD-SHELL", "curl -f ${var.health_check_url} || exit 1"
       ],
     },
     logConfiguration : {
       logDriver : "awslogs",
       options : {
         awslogs-region : var.aws_region,
-        awslogs-group : "droits-backoffice-container-logs",
-        awslogs-stream-prefix : "backoffice"
+        awslogs-group : "droits-${var.service_name}-container-logs",
+        awslogs-stream-prefix : var.service_name
       }
     }
   }])
 }
 
-resource "aws_ecs_service" "backoffice-service" {
-  name                              = "backoffice"
-  cluster                           = aws_ecs_cluster.droits-ecs-cluster.id
-  task_definition                   = aws_ecs_task_definition.backoffice-task-definition.arn
+resource "aws_ecs_service" "service" {
+  name                              = var.service_name
+  cluster                           = var.droits_ecs_cluster
+  task_definition                   = aws_ecs_task_definition.task-definition.arn
   launch_type                       = "FARGATE"
   desired_count                     = 1
```

Comment:
maybe pull out desired_count & health_check_grace_period_seconds as vars too, fine as 1 & 600 for now, but we might wanna have 2 backends etc at some point

Reply:
Done